internal/tests/testrepository/header_testrepository.go

```diff
@@ -193,6 +194,49 @@
 	return headers, nil
 }
 
+func (r *HeaderTestRepository) GetMerkleRoots(batchSize int, lastEvaluatedKey int) (*domains.MerkleRootsESKPagedResponse, error) {
+	merkleroots := make([]*domains.BlockHeader, 0)
+	orderByField := "BlockHeight"
+	sortDirection := "ASC"
+	// order headers by height ASC
+	sort.Slice(*r.db, func(i, j int) bool {
+		return (*r.db)[i].Height < (*r.db)[j].Height
+	})
+	// include only those headers that height is greater than lastEvaluatedKey
+	for _, header := range *r.db {
+		if header.Height > int32(lastEvaluatedKey) {
```

Comment:
Here it shouldn't be a problem since the Height is database field of type int32 so it shouldn't be an issue

Reply:
I know 🙈 - but try, maybe, switch the conversion:
```golang
if int(header.Height) > lastEvaluatedKey {
```

**Explanation:**
Casting `header.Height` to `int`: By using `int(header.Height)`, you're converting the `int32 ` value to an `int` so that it can be compared with `lastEvaluatedKey`, which is already of type `int`. Since `int32` is always smaller or equal in size to `int` (on platforms where int is at least 32 bits), this cast is safe in terms of avoiding overflows.

Reply:
applied